returnn/datasets/basic.py

```diff
@@ -1304,6 +1330,22 @@ def index_shape_for_batches(cls, batches, data_key="data"):
             shape = [max(shape[0], batch.max_num_frames_per_slice[data_key]), shape[1] + batch.num_slices]
         return tuple(shape)
 
+    @classmethod
+    def extend_default_kwargs_from_parent_dataset(
```

Comment:
This should be private, as this is only intended for internal purpose.

But also, I think I prefer to not have this as a member function. We already have too much here.